scipy/stats/tests/test_stats.py

```diff
@@ -2229,7 +2333,7 @@ def test_ttest_rel():
         # check that nan in input array result in nan output
         anan = np.array([[1, np.nan], [-1, 1]])
         assert_equal(stats.ttest_rel(anan, np.zeros((2, 2))),
-                     ([0, np.nan], [1, np.nan]))
+                     (np.nan, np.nan))
```

Comment:
Good catch. The change isn't right. This demonstrates that the behavior for nan propagation currently in master works fine as is:

    In [1]: from scipy import stats

    In [2]: a2 = stats.norm.rvs(size=9).reshape(3,3)

    In [3]: b2 = np.arange(9).reshape(3,3)

    In [4]: stats.ttest_rel(a2, b2)
    Out[4]: Ttest_relResult(statistic=array([-1.44542944, -1.70106168, -1.93403464]), pvalue=array([ 0.28521702,  0.23103717,  0.19278399]))

    In [5]: a2[0,0] = np.nan

    In [6]: stats.ttest_rel(a2, b2)   # default is axis=0, so only expect a nan for the first axis here
    Out[6]: Ttest_relResult(statistic=array([        nan, -1.70106168, -1.93403464]), pvalue=array([        nan,  0.23103717,  0.19278399]))

That's why I just commented about the 2 lines to be removed from ``ttest_rel``. And so the change to the test here needs to be undone.